src/days/day03.rs

```diff
@@ -70,36 +66,36 @@ fn part1(input: Map, slope: Slope) -> u32 {
             }
         };
 
-        p = apply_slope(slope, p)
+        p.apply_slope(&slope);
     }
 }
 
 
 #[cfg(test)]
 mod tests {
     use std::fs;
-    use days::day03::{Type, Map, part1};
-    use itertools::Itertools;
+    use days::day03::{Type, Map, Position, part1};
 
     fn get_input() -> Map {
         let input = fs::read_to_string("input/03.txt").expect("Unable to open file");
-        let rows = input.trim().split('\n').into_iter().map(|s| s.trim()).map(|s| {
-            Box::new(s.chars().into_iter().map(|c| Type::from_char(c)).collect_vec())
-        }).collect_vec();
+        let rows = input
+            .trim()
+            .lines()
+            .map(str::trim)
+            .map(|s| s.chars().map(Type::from_char).collect())
+            .collect();
```

Comment:
* `.lines()` is roughly the same as `.split('\n').into_iter()`
* We can simplify some calls by passing the functions directly, rather than using closures
* Boxing isn't necessary here, and we can just use .collect()